### PR TITLE
Use default unstructured converter

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:

Using `runtime.DefaultUnstructuredConverter` is recommended way to operate upon unstructured objects while using controller-runtime.  It also makes code more readable

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Review by alexander-demicev — APPROVED** [review]
nice!

**Review by mjura — APPROVED** [review]
LGTM

File: internal/rancher/cluster.go [hunk 1]
```diff
@@ -40,43 +27,21 @@ type ClusterStatus struct {
...
-
-	if err := decoder.Decode(r); err != nil {
-		return nil, fmt.Errorf("failed to decode rancher cluster: %w", err)
+		return nil, fmt.Errorf("failed to decode cluster: %w", err)
 	}
 
+	rancherClusterUnstructured := &unstructured.Unstructured{}
+	rancherClusterUnstructured.SetUnstructuredContent(obj)
 	rancherClusterUnstructured.SetGroupVersionKind(gvkRancherCluster)
-	rancherClusterUnstructured.SetCreationTimestamp(metav1.Now())
```
Comment (furkatgofurov7) [inline internal/rancher/cluster.go:59]: curious, why is this not needed anymore?

Comment (alexander-demicev) [reply internal/rancher/cluster.go:59]: I had some issues previously with missing creation timestamp when converting from unstructured, I guessed it's fixed now when using `runtime.DefaultUnstructuredConverter`

**Pull request merged**